### ISSUE TITLE
Training details about generating pseudo groundtruth

### ISSUE THREAD

**Issue by lianqing11** [clovaai/CRAFT-pytorch#30]
Hi, thanks for your work. I have some questions about the detail of generating pseudo ground-truth.

1. In Figure 6, I am confused that why can't we just input the whole image to generate the pseudo ground truth (non-text region area are padded with 0).

2. How to crop the image?
    As I understand for the paper Figure 4, we should first crop the text region and then feed them to the network. Is that right? However, in the ICDAR2015, the word ground truths aren't the normal rectangle. 1) How do we crop it? (rotate it and then crop?).

**Comment by wulitaotao1** [comment]
i think that when the confidence of  the pseudo gt lower than 0.5, its not easy to average the minAreaRect, so i first take perspective transform to the ground truths,then inference and watershed labeling,finally get inverse perspective transform and reflect to the original img.

**Comment by lianqing11** [comment]
@wulitaotao1 Thanks for your reply. I don't quite understand what kind of transform should we take. Do you mean rotation?

**Comment by Godricly** [comment]
1. In #18 you can see that full image forwarding is possible. But still you need word bounding box to generate Gaussian maps.
2. They are several approaches to crop word patch: MinRect, affine mapping, perspective transform.

**Comment by YoungminBaek** [comment]
@lianqing11

1. The purpose of using cropped images for generating the pseudo ground truth is to measure the error of word length. You know how many characters are in the word image, but not the whole image.
For that scenario, it is possible to inference the whole input image and generate the character bounding boxes by cropping the score map for each word box as @Godricly mentioned.

2. We use the perspective transform to crop the image. If you have at least four corresponding points, you can find the transformation matrix between two different coordinates.
Please refer to the https://docs.opencv.org/2.4/modules/imgproc/doc/geometric_transformations.html#getperspectivetransform.

**Comment by lianqing11** [comment]
@Godricly  Thanks for your advice. I have tried using the full image to generate the pseudo label. However, the improvement is limited. I only could get 7% h-mean improvement on ICDAR15 dataset (from 60% to 67%).  Have you tried this method? I don't know if this would harm the performance.

**Comment by Godricly** [comment]
Nope, I ignored all words cropped by the cropping box to boost training speed.
